### PR TITLE
[cmocka] Add builder

### PR DESCRIPTION
**Pull request by bramtayl**
This is just a draft PR so I can ask a question

### CODE REVIEW

**Comment by bramtayl** [comment]
I'm getting 

```
CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
```

Should `CMAKE_MAKE_PROGRAM` be an environmental variable? Or in `CMAKE_TARGET_TOOLCHAIN`?

**Comment by bramtayl** [comment]
Also, a road-map of how I got here: cmocka -> libavtp -> alsa-plugins -> PortAudio

File: C/cmocka/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "cmocka"
+version = v"1.1.5"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://cmocka.org/files/1.1/cmocka-1.1.5.tar.xz", "f0ccd8242d55e2fd74b16ba518359151f6f8383ff8aef4976e48393f77bba8b6"),
```
Comment (giordano) [inline C/cmocka/build_tarballs.jl:10]:
```suggestion
    ArchiveSource("https://cmocka.org/files/1.1/cmocka-1.1.5.tar.xz", "f0ccd8242d55e2fd74b16ba518359151f6f8383ff8aef4976e48393f77bba8b6"),
```
probably?

File: C/cmocka/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+sources = [
+    ArchiveSource("https://cmocka.org/files/$(version.major).$(version.minor)/cmocka-$(version).tar.xz", "f0ccd8242d55e2fd74b16ba518359151f6f8383ff8aef4976e48393f77bba8b6"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/cmocka-*/
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
```
Comment (giordano) [inline C/cmocka/build_tarballs.jl:18]:
```suggestion
cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    ..
```

**Comment by giordano** [comment]
> This is just a draft PR so I can ask a question
>
> I'm getting
> ```
> CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
> ```


Where? :raised_eyebrow:

**Comment by bramtayl** [comment]
Oh huh, it was locally, but doesn't seem to be an issue here yay! The first musl warning points to

```
/* ILP32 and LP64 platforms */
#  ifdef __WORDSIZE /* glibc */
#   if __WORDSIZE == 64
      typedef unsigned long int uintptr_t;
#   else
      typedef unsigned int uintptr_t;
#   endif /* __WORDSIZE == 64 */
#  else /* __WORDSIZE */
#   if defined(_LP64) || defined(_I32LPx)
      typedef unsigned long int uintptr_t;
#   else
      typedef unsigned int uintptr_t;
#   endif
#  endif /* __WORDSIZE */
# endif /* _WIN32 */
```

So looks like we need to define `__WORDSIZE` `_LP64` or `_I32LPx`?

**Comment by bramtayl** [comment]
I'll try adding the two patches from alpine

File: C/cmocka/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+sources = [
+    ArchiveSource("https://cmocka.org/files/$(version.major).$(version.minor)/cmocka-$(version).tar.xz", "f0ccd8242d55e2fd74b16ba518359151f6f8383ff8aef4976e48393f77bba8b6"),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/cmocka-*/
+atomic_patch -p1 $WORKSPACE/srcdir/patches/wordsize.patch
+atomic_patch -p1 $WORKSPACE/srcdir/patches/musl_uintptr.patch
```
Comment (giordano) [inline C/cmocka/build_tarballs.jl:18]:
```suggestion
if [[ "${target}" = *-linux-musl* ]]; then
    atomic_patch -p1 $WORKSPACE/srcdir/patches/wordsize.patch
    atomic_patch -p1 $WORKSPACE/srcdir/patches/musl_uintptr.patch
fi
```

File: C/cmocka/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = LibraryProduct[
+    # TBD
+]
```
Comment (giordano) [inline C/cmocka/build_tarballs.jl:38]:
```suggestion
products = [
    LibraryProduct("cmocka", :cmocka),
]
```

File: C/cmocka/build_tarballs.jl [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("cmocka", :cmocka),
```
Comment (giordano) [inline C/cmocka/build_tarballs.jl:37]:
```suggestion
    LibraryProduct(["libcmocka", "cmocka"], :libcmocka),
```

**Comment by bramtayl** [comment]
Thanks!

**Pull request merged**